### PR TITLE
session: set Sleep state for process info

### LINKED ISSUE #7825: mydumper refuses to backup with idle sessions
## Bug Report

Please answer these questions before submitting your issue. Thanks!

1. What did you do?

It looks like mydumper breaks in a lot of cases due to https://github.com/pingcap/tidb/pull/7765 now including an accurate processlist time.  It incorrectly triggers mydumper's test to make sure there are no long running transactions.

2. What did you expect to see?

The problem is that idle sessions in TiDB are said to be in the Command of "Query" whereas in MySQL they would be in the Command of "Sleep".  i.e.

```
MySQL [(none)]> show processlist;
+------+------+-----------+-----------+---------+------+-------+------------------+------+
| Id   | User | Host      | db        | Command | Time | State | Info             | Mem  |
+------+------+-----------+-----------+---------+------+-------+------------------+------+
|   81 | root | 127.0.0.1 |           | Query   |    0 | 2     | show processlist |    0 |
|   44 | root | 127.0.0.1 | bikeshare | Query   |   99 | 2     |                  |    0 |
+------+------+-----------+-----------+---------+------+-------+------------------+------+
2 rows in set (0.00 sec)
```

There is no activity, it should be safe to backup.

3. What did you see instead?

This caused the following error until I 'pinged' the session to reset the idle time.

```
tocker@penguin:~/go/src/github.com/morgo/mydumper$ ./mydumper -v 3  --database bikeshare
** Message: Server version reported as: 5.7.10-TiDB-v2.1.0-rc.2-38-g081920da9-dirty
** Message: Connected to a TiDB server

** (mydumper:12392): CRITICAL **: There are queries in PROCESSLIST running longer than 60s, aborting dump,
        use --long-query-guard to change the guard value, kill queries (--kill-long-queries) or use
        different server for dump
```

4. What version of TiDB are you using (`tidb-server -V` or run `select tidb_version();` on TiDB)?
```
MySQL [(none)]> select tidb_version()\G
*************************** 1. row ***************************
tidb_version(): Release Version: v2.1.0-rc.2-38-g081920da9-dirty
Git Commit Hash: 081920da935e04967df1c408841e013d010eb1b3
Git Branch: showtablestatus
UTC Build Time: 2018-09-27 12:59:15
GoVersion: go version go1.11 linux/amd64
Race Enabled: false
TiKV Min Version: 2.1.0-alpha.1-ff3dd160846b7d1aed9079c389fc188f7f5ea13e
Check Table Before Drop: false
1 row in set (0.00 sec)
```

### PR DESCRIPTION
**Pull request by coocood**
### What problem does this PR solve? 
Fixes #7825

### What is changed and how it works?
Set the process command to `Sleep` if `sql` argument is empty

### Check List 

Tests 

 - Unit test

Related changes

 - Need to cherry-pick to the release branch
 - Need to be included in the release note

### CODE REVIEW

File: session/session.go [hunk 1]
```diff
@@ -712,6 +712,9 @@ func (s *session) SetProcessInfo(sql string) {
 		State:   s.Status(),
 		Info:    sql,
 	}
+	if sql == "" {
```
Comment (shenli) [inline session/session.go]: If the query is sent through the binary protocol with prepare-execute method, will the `sql` be an empty string?

Comment (jackysp) [reply session/session.go]: No, `SetProcessInfo` will use `OriginText` of the execute statement. It is set in `CompileExecutePreparedStmt`.

Comment (shenli) [reply session/session.go]: OK

**Comment by shenli** [comment]
LGTM

**Pull request merged**